app/src/main/java/helium314/keyboard/latin/Suggest.kt

```diff
@@ -395,17 +396,20 @@ class Suggest(private val mDictionaryFacilitator: DictionaryFacilitator) {
             wordComposer: WordComposer, results: SuggestionResults,
             trailingSingleQuotesCount: Int, defaultLocale: Locale
         ): ArrayList<SuggestedWordInfo> {
+            val keyboardShiftMode = KeyboardSwitcher.getInstance().keyboardShiftMode
```

Comment:
Please use the the `keyboard` provided in `getSuggestedWordsForNonBatchInput` for consistency (I want to avoid weird edge cases where there could be 2 different keyboards).
I just moved the `mElementId` -> shift / caps mode conversion into `KeyboardId` so you don't need the `KeyboardSwitcher` instance.

Reply:
I think I had a reason for making it static, but can't figure it out. Anyway, if necessary I can change it back again.